core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -964,7 +964,7 @@ public TerminationHandle warmTopReader(final IndexShard indexShard, IndexMetaDat
             final Map<String, MappedFieldType> warmUpGlobalOrdinals = new HashMap<>();
             for (DocumentMapper docMapper : mapperService.docMappers(false)) {
                 for (FieldMapper fieldMapper : docMapper.mappers()) {
-                    final FieldDataType fieldDataType = fieldMapper.fieldType().fieldDataType();
+                    final FieldDataType fieldDataType = fieldMapper.fieldDataType();
```

Comment:
Won't the `indexName` be wrong below, since it is just `_parent`? Maybe we should just have a special case here for `_parent` for now? Something like:
```
String indexName = fieldMapper.fieldType.names().indexName();
FieldDataType fieldDataType = fieldMapper.fieldType().fieldDataType();
if (fieldMapper instanceOf ParentFieldMapper) {
    ParentFieldMapper parentMapper = (ParentFieldMapper)fieldMapper;
    indexName = parentMapper.getJoinFieldType().names().indexName();
    fieldDataType = parentMapper.getJoinFieldType().fieldDataType();
}
```